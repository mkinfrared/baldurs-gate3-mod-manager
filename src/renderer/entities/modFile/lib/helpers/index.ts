export { extractPakFile } from "./extractPakFile";

export { getModFiles } from "./getModFiles";

export { getModInfo } from "./getModInfo";

export { readZipInfo } from "./readZipInfo";
