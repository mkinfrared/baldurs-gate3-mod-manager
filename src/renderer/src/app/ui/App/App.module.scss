.App {
  display: grid;
  grid-template-columns: auto 1fr;
  min-height: 100dvh;

  main {
    display: grid;
    gap: var(--spacing-10);
    grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
    padding: var(--spacing-10);
  }
}
