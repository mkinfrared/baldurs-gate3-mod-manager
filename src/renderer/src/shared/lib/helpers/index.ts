export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { trpcClient, trpc } from "./trpc";

export * from "./typeGuards";
