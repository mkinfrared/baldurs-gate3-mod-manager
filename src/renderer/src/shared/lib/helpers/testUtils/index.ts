export { renderWithRouter, withQueryProvider } from "./testUtils";
