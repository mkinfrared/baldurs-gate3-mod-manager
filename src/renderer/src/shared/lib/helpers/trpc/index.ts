export { trpc, trpcClient } from "./trpc";
