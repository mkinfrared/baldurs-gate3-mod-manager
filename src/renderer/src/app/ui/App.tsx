import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { MemoryRouter as Router } from "react-router-dom";

import { AppRoutes } from "@renderer/routes";
import { trpc, trpcClient } from "@renderer/shared/lib/helpers";
import { AppNavBar } from "@renderer/widgets/ui";

import css from "./App.module.scss";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      suspense: true,
    },
  },
});

/**
 * https://trpc.io/docs/v9/react
 * https://github.com/awohletz/electron-prisma-trpc-example/blob/main/src/client/renderer.ts
 */

const App = () => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>
      <Router>
        <div className={css.App} data-testid="App">
          <aside>
            <AppNavBar className={css.navBar} />
          </aside>
          <main>
            <AppRoutes />
          </main>
        </div>
      </Router>
    </QueryClientProvider>
  </trpc.Provider>
);

export { App };

export default App;
