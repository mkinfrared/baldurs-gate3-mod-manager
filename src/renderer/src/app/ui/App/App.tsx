import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { MemoryRouter as Router } from "react-router-dom";

import { AppRoutes } from "@renderer/routes";
import { trpc, trpcClient } from "@renderer/shared/lib/helpers";
import { AppNavBar } from "@renderer/widgets/ui";

import css from "./App.module.scss";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      suspense: true,
    },
  },
});

const App = () => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>
      <Router>
        <div className={css.App}>
          <aside>
            <AppNavBar />
          </aside>
          <main>
            <AppRoutes />
          </main>
        </div>
      </Router>
    </QueryClientProvider>
  </trpc.Provider>
);

export { App };

export default App;
