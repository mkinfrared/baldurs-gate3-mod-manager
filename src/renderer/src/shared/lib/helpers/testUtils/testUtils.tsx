/* eslint-disable import/no-unused-modules */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { ReactElement, ReactNode } from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import { trpc, trpcClient } from "@renderer/shared/lib/helpers/trpc";

// https://testing-library.com/docs/example-react-router/
const renderWithRouter = (ui: ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

const queryClient = new QueryClient();

const withQueryProvider = (children: ReactNode) => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </trpc.Provider>
);

const withRouter = (children: ReactNode) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export { renderWithRouter, withQueryProvider, withRouter };
