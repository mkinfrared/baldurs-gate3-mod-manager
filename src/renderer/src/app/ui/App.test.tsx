import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { withQueryProvider } from "@renderer/shared/lib/helpers/testUtils";

import { App } from "./App";

vi.mock("@renderer/shared/lib/hooks");

describe("<App/>", () => {
  const Component = <App />;

  it("should be defined", () => {
    expect(App).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(withQueryProvider(Component));

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(withQueryProvider(Component));
    const element = getByTestId("App");

    expect(element).toBeDefined();
  });
});
