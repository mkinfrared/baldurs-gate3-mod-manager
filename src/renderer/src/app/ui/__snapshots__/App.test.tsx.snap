// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App/> > should match the snapshot 1`] = `
<div>
  <div
    class="App"
    data-testid="App"
  >
    <aside>
      <div
        class="AppNavBar navBar"
        data-testid="AppNavBar"
      >
        <a
          href="/baldurs-gate-3"
        >
          <img
            alt="Baldur's Gate 3 Logo"
            src="/src/renderer/src/shared/assets/images/BG3-logo.png"
          />
        </a>
      </div>
    </aside>
    <main>
      <div
        class="BaldursGate3"
        data-testid="BaldursGate3"
      >
        <div
          class="InstalledModsSection"
          data-testid="InstalledModsSection"
        >
          <div>
            Loading..
          </div>
        </div>
        <div
          class="ModFileSection"
          data-testid="ModFileSection"
        >
          <div
            class="StickyBlock"
            data-testid="StickyBlock"
          >
            <h3
              class="Heading"
              data-testid="Heading"
            >
              Modification Files
            </h3>
          </div>
          <div
            class="ModFileList"
            data-testid="ModFileList"
          >
            <div
              class="Card ModFileRow header"
              data-testid="Card"
            >
              <span>
                Modification Name
              </span>
              <span>
                Version
              </span>
              <button
                class="Button ModFileButton contained primary"
                data-testid="ModFileButton"
                type="button"
              >
                <svg
                  class="Spinner spinner"
                >
                  bars.svg
                </svg>
                <span
                  class="content"
                >
                  Install
                </span>
              </button>
            </div>
            <div
              class="Card EmptyList"
              data-testid="Card"
            >
              <h4
                class="Heading header"
                data-testid="Heading"
              >
                Drag and drop you 
                zip, rar, 7z
                 files here or
              </h4>
              <button
                class="Button contained primary"
                data-testid="Button"
                type="button"
              >
                <svg
                  class="Spinner spinner"
                >
                  bars.svg
                </svg>
                <span
                  class="content"
                >
                  <label
                    for="file"
                  >
                    Click this button to select files
                  </label>
                </span>
              </button>
              <input
                accept="zip,rar,7z"
                id="file"
                multiple=""
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
